### PR TITLE
JDK-8277382 make c1 BlockMerger use IR::verify only when necessary

### PR DESCRIPTION
**Pull request by LudwikJaniuk**
This PR removes two calls to `IR::verify` which were unnecessary. The reason they are unnecessary is that `try_merge` does not always take any action. There is not need to verify if nothing has changed. In the cases that `try_merge` does do anything, it already calls `IR::verify` afterwards.

This PR also switches some deeply nested if statements in `try_merge` to early returns.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277382](https://bugs.openjdk.java.net/browse/JDK-8277382): make c1 BlockMerger use IR::verify only when necessary

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6456/head:pull/6456` \
`$ git checkout pull/6456`

Update a local copy of the PR: \
`$ git checkout pull/6456` \
`$ git pull https://git.openjdk.java.net/jdk pull/6456/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6456`

View PR using the GUI difftool: \
`$ git pr show -t 6456`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6456.diff">https://git.openjdk.java.net/jdk/pull/6456.diff</a>

</details>

### CODE REVIEW

**Comment by LudwikJaniuk** [comment]
Passes both tier1 and tier2 in debug build.

File: src/hotspot/share/c1/c1_Optimizer.cpp [hunk 1]
```diff
@@ -336,140 +336,135 @@ class BlockMerger: public BlockClosure {
...
+    Value sux_value;
+    for_each_stack_value(sux_state, index, sux_value) {
+      assert(sux_value == end_state->stack_at(index), "stack not equal");
+    }
+    for_each_local_value(sux_state, index, sux_value) {
+        Phi* sux_phi = sux_value->as_Phi();
+        if (sux_phi != NULL && sux_phi->is_illegal()) continue;
+        assert(sux_value == end_state->local_at(index), "locals not equal");
+      }
+    assert(sux_state->caller_state() == end_state->caller_state(), "caller not equal");
```
Comment (TobiHartmann) [inline src/hotspot/share/c1/c1_Optimizer.cpp:374]: The indentation is wrong.

**Review by TobiHartmann — CHANGES_REQUESTED** [review]

File: src/hotspot/share/c1/c1_Optimizer.cpp (hunk 1 above)
Comment (LudwikJaniuk) [reply src/hotspot/share/c1/c1_Optimizer.cpp]: done

**Review by neliasso — APPROVED** [review]
Looks good!

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Pull request closed without merging**